src/shapes/ellipsoidsmesh.cpp

```diff
@@ -0,0 +1,498 @@
+#include <mitsuba/core/fwd.h>
+#include <mitsuba/core/fresolver.h>
+#include <mitsuba/core/properties.h>
+#include <mitsuba/core/util.h>
+#include <mitsuba/core/timer.h>
+#include <mitsuba/render/mesh.h>
+
+#include "ellipsoids.h"
+
+NAMESPACE_BEGIN(mitsuba)
+
+/// Forward declaration of template shell mesh data buffers
+extern std::vector<dr::Array<float, 3>>    box_vertices, uv_sphere_72_vertices, ico_sphere_20_vertices;
+extern std::vector<dr::Array<uint32_t, 3>> box_faces,    uv_sphere_72_faces,    ico_sphere_20_faces;
+
+/**!
+
+.. _shape-EllipsoidsMesh:
+
+Mesh ellipsoids (:monosp:`EllipsoidsMesh`)
```

Comment:
I think that "mesh ellipsoids" is a nice name to refer to these, but EllipsoidsMesh sounds kind of odd.
Can you please do a global rename of
- "EllipsoidsMesh" -> "MeshEllipsoids"
- "ellipsoids_mesh" -> "mesh_ellipsoids"?
- "ellipsoidsmesh" -> "mesh_ellipsoids"

Reply:
The reason why I did not use the underscore here was to be consistent with the other plugins (e.g. `linearcurve`, `shapegroup`, ...).

Also I like `EllipsoidsMesh` better as puts an emphasis on the `Ellipsoids` part, which makes the code more readible IMO.